py/dml/codegen.py

```diff
@@ -2399,53 +2430,66 @@ def stmt_assign(stmt, location, scope):
 
         stmts = []
         lscope = Symtab(scope)
-        syms = []
+        stmt_pairs = []
         for (i, (tgt, src_ast)) in enumerate(zip(tgts, src_asts)):
-            init = eval_initializer(site, tgt.ctype(), src_ast, location,
-                                    scope, False)
-            name = 'tmp%d' % (i,)
-            sym = lscope.add_variable(
-                    name, type=tgt.ctype(), site=tgt.site, init=init,
-                    stmt=True)
-            syms.append(sym)
-
-        stmts.extend(map(sym_declaration, syms))
-        stmts.extend(
-            mkAssignStatement(
-                tgt.site, tgt, ExpressionInitializer(mkLocalVariable(tgt.site,
-                                                                     sym)))
-            for (tgt, sym) in zip(tgts, syms))
-        return stmts
+            if isinstance(tgt, NonValue):
+                expr = codegen_src_for_nonvalue_target(site, tgt, src_ast,
+                                                       location, scope)
+                stmt_pairs.append((mkCopyData(tgt.site, expr, tgt), None))
+            else:
+                init = eval_initializer(site, tgt.ctype(), src_ast, location,
+                                        scope, False)
+                name = 'tmp%d' % (i,)
```

Comment:
`_tmp%d`? (because what if `extern int tmp0(void);` exists and the initializer of a subsequent assignment calls it)

Reply:
oh, the problem was not introduced by you. Still valid but not blocking then.

Reply:
I just want to note that any issue this could only possibly is incredibly niche. In particular, this could only present an issue if `-g` is switched on, because otherwise the C variable name will get uniquified. (Notice that all codegen involved here uses `scope` and not `lscope`, so DML resolution will never take these temporary variables into account; the only possible issue is if the wrong thing happens at C level).

But yes; if `-g` is switched on, and the initializer references some C literal `tmp0` (made accessible via `extern`), then this could present an issue. But it's rather far-fetched.